src/rust/ensogl/src/display/scene.rs

```diff
@@ -702,28 +702,44 @@ impl ViewData {
 // =============
 
 /// Please note that currently the `Views` structure is implemented in a hacky way. It assumes the
-/// existence of `main`, `cursor`, and `label` views, which are needed for the GUI to display shapes
-/// properly. This should be abstracted away in the future.
+/// existence of `main`, `overlay`, `cursor`, and `label` views, which are needed for the GUI to
+/// display shapes properly. This should be abstracted away in the future.
 #[derive(Clone,CloneRef,Debug)]
 pub struct Views {
-    logger     : Logger,
-    pub main   : View,
-    pub cursor : View,
-    pub label  : View,
-    all        : Rc<RefCell<Vec<WeakView>>>,
-    width      : f32,
-    height     : f32,
+    logger                    : Logger,
+    pub main                  : View,
+    pub cursor                : View,
+    pub label                 : View,
+    pub overlay               : View,
+    pub visualisation         : View,
+    pub overlay_visualisation : View,
+    pub overlay_cursor        : View,
+    all                       : Rc<RefCell<Vec<WeakView>>>,
+    width                     : f32,
+    height                    : f32,
 }
 
 impl Views {
     pub fn mk(logger:&Logger, width:f32, height:f32) -> Self {
-        let logger = logger.sub("views");
-        let main   = View::new(&logger,width,height);
-        let cursor = View::new(&logger,width,height);
-        let label  = View::new_with_camera(&logger,&main.camera);
-        let all    = vec![main.downgrade(),cursor.downgrade(),label.downgrade()];
-        let all    = Rc::new(RefCell::new(all));
-        Self {logger,main,cursor,label,all,width,height}
+        let logger                 = logger.sub("views");
+        let main                   = View::new(&logger,width,height);
+        let cursor                 = View::new(&logger,width,height);
+        let visualisation          = View::new(&logger,width,height);
+        let overlay                = View::new(&logger,width,height);
+        let overlay_visualisation  = View::new(&logger,width,height);
+        let overlay_cursor         = View::new(&logger,width,height);
+        let label                  = View::new_with_camera(&logger,&main.camera);
+        let all                    = vec![main.downgrade(),
+                                          visualisation.downgrade(),
+                                          cursor.downgrade(),
+                                          label.downgrade(),
+                                          overlay.downgrade(),
+                                          overlay_visualisation.downgrade(),
+                                          overlay_cursor.downgrade(),
+                                     ];
```

Comment:
Break alignment and make it 2 lines instead pls :)

Reply:
I think this one make more sense vertical: the layer ordering indicates the occlusion of the layers. Scanning it top to bottom here and seeing the order is easier than if it is inline.